data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/consumer/KafkaCustomConsumer.java

```diff
@@ -424,16 +426,25 @@ private void processRecord(final AcknowledgementSet acknowledgementSet, final Re
                 bufferAccumulator.add(record);
                 break;
             } catch (Exception e) {
+                if (!paused) {
+                    paused = true;
+                    consumer.pause(consumer.assignment());
+                }
                 if (e instanceof SizeOverflowException) {
                     topicMetrics.getNumberOfBufferSizeOverflows().increment();
                 } else {
                     LOG.debug("Error while adding record to buffer, retrying ", e);
                 }
                 try {
                     Thread.sleep(100);
+                    consumeRecords();
```

Comment:
Instead of conumeRecords() should we call consumer.poll() here directly ? and if it returns records then we assert and may be restart consumer because poll returning records means there is a bug somewhere. Calling consumeRecords() will re-enter this function if poll() returns messages.

Reply:
That's fair. I can just do poll() and assert on records returned.